File: finanze/application/ports/auto_contributions_port.py
import abc
from uuid import UUID

from domain.auto_contributions import AutoContributions, ContributionQueryRequest
from domain.entity import Entity
from domain.fetch_record import DataSource


class AutoContributionsPort(metaclass=abc.ABCMeta):
    @abc.abstractmethod
    async def save(self, entity_id: UUID, data: AutoContributions, source: DataSource):
        raise NotImplementedError

    @abc.abstractmethod
    async def get_all_grouped_by_entity(
        self, query: ContributionQueryRequest
    ) -> dict[Entity, AutoContributions]:
        raise NotImplementedError

    @abc.abstractmethod
    async def delete_by_source(self, source: DataSource):
        raise NotImplementedError

File: mobile/src/application/ports/core/autoContributionsPort.ts
import {
  AutoContributions,
  ContributionQueryRequest,
  DataSource,
  Entity,
} from "@/domain"

export interface AutoContributionsPort {
  save(
    entityId: string,
    data: AutoContributions,
    source: DataSource,
  ): Promise<void>
  getAllGroupedByEntity(
    query: ContributionQueryRequest,
  ): Promise<Map<Entity, AutoContributions>>
  deleteBySource(source: DataSource): Promise<void>
}
